pkg/controller/external_nofork.go

```diff
@@ -286,6 +335,29 @@ func (n *noForkExternal) getResourceDataDiff(ctx context.Context, s *tf.Instance
 			return nil, errors.Wrap(err, "failed to compute the customized terraform.InstanceDiff")
 		}
 	}
+	if !instanceDiff.Empty() && resourceExists {
```

Comment:
Let's refactor this filtering logic further into its own function.

Reply:
We are doing so in the [next PR](https://github.com/ulucinar/upbound-upjet/pull/6).